src/platform/input/sdl2mouse.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief SDL2 Mouse Input interface.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#include "sdl2mouse.h"
+#include "editmain.h"
+#include "gameclient.h"
+#include <cstdio>
+
+SDL_Cursor *SDL2Mouse::s_loadedCursors[CURSOR_COUNT][8];
+
+SDL2Mouse::SDL2Mouse() : m_nextFreeIndex(0), m_nextGetIndex(0) {}
+
+void SDL2Mouse::Init()
+{
+    Mouse::Init();
+
+    std::memset(m_eventBuffer, 0, sizeof(m_eventBuffer));
+    std::memset(s_loadedCursors, 0, sizeof(s_loadedCursors));
```

Comment:
In `Win32Mouse` class the memset is in constructor. Any reason this is put in `Init` function here instead?